lnwire/custom.go

```diff
@@ -20,10 +69,12 @@ type Custom struct {
 // interface.
 var _ Message = (*Custom)(nil)
 
-// NewCustom instanties a new custom message.
+// NewCustom instantiates a new custom message.
 func NewCustom(msgType MessageType, data []byte) (*Custom, error) {
-	if msgType < CustomTypeStart {
-		return nil, errors.New("msg type not in custom range")
+	if msgType < CustomTypeStart && !IsCustomOverride(msgType) {
```

Comment:
Seems like an alternative path here would be to:
  * add a functional option arg to `NewCustom`
  * let that pass in the set of custom override types
  * do this check based on that 

I think that would let us get rid of the current global that isn't safe for concurrent access?

Reply:
> pass around the list of allowed overrides everywhere then? That might be a bit tricky (and add lots of refactoring)

Yeah, this is the problem with trying to move away from a global - lots of lift in various parts of the code base.

Reply:
One way to do this would be to create a new struct something like `CustomMsgWriter/Reader`. Then that can accept the set of new custom message types that should be permitted. Relevant call sites that need to be able to handle them would then instantiate this, and use it instead of the direct `lnwire.Read/WriteMessage` calls. `lnwire.WriteMessage` and the like can then gain the ability to forcibly read/write a custom message out of te range via an optional arg. With this path, there's no global needed and the call sites that don't necessarily need to handle this new custom msg range don't change (functional arg so no ned for them to pass the extra message).